front: put speed space chart into board wrapper

iLTmpas opened the pull request:
close #12593

See commits
 
:warning: A significant change has been made to the diagram: it no longer controls its own size but takes its size as an argument from its parent component. This change was necessary in order to implement a wrapper that changes its size.

All the logic around managing the size of the diagram and the height at which to draw the layers has therefore changed. A new `internalHeight` has been introduced, which corresponds to the height of the graph with only the curves + the TOP and BOTTOM margins of the graph.

In a future PR, we will remove the margins from this variable to retain only the height of the graph with the curves.

SharglutDev commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -252,26 +253,31 @@ const SimulationResults = ({
-              />
+          <BoardWrapper
+            hidden={!activeBoards.has('sdd')}
+            name={t('simulationResults.speedSpaceChart')}
```
On the mockup, the board title in english is `SDD` in abbreviation. So probably the french name should be `GEV` as well (same as in the scenario header).

SharglutDev commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -252,26 +253,31 @@ const SimulationResults = ({
+                  height: `${speedSpaceChartContainerHeight + HANDLE_TAB_RESIZE_HEIGHT}px`,
+                }}
+              >
+                <SpeedSpaceChartContainer
```
Something seems to be wrong compared to the mockup. The space below the SDD is too large.

It comes from : 
- the SDD height. On the app it's default height is 521.5px but in the mockup it's 460px (without the 22px of the board header)
- The draggable part which adds some space below the SDD container. In the mockup, the draggable button is outside of the board.

Also the options button are not disabled on the chart anymore. To be confirmed with @thibautsailly but my guess is that they are now inside the board menu at the top right.

thibautsailly replied:
Could you add a screenshot so I can see what you're referring to? Thank you!

SharglutDev replied:
<img width="1104" height="564" alt="Capture d’écran 2025-07-31 à 11 03 37" src="https://github.com/user-attachments/assets/46ddda1b-50bd-46e5-8fa1-e748029928d8" />

SharglutDev commented:
Thank you for the changes. Osrd-ui changes should not be in the same commit as te ones in osrd. Please move the ui changes in a separate commit starting with `ui-charts: ...`

iLTmpas committed:
ui-charts: update speed-space-chart height in stories

Signed-off-by: Louis Tran <louistran29112003@gmail.com>

SharglutDev reviewed (commented):
I think the PR is good enough for now.

As you can see when comparing the [old SDD UI](https://www.sketch.com/s/2fb064cd-c483-46ac-bd30-72705fd63e75/p/9559F6FC-69EE-4DB8-BE1A-77F60F10C97F/canvas) and the [new one](https://www.sketch.com/s/008d6980-d719-4bdc-bb2a-b4c2dda0dc1e/p/8017F0A2-FC69-4111-A2C6-A38E6D37AB65/canvas), a lot has changed in the UI.
Theses changes should be taken into account to match the new scenario page design.

I'm ok to merge this PR on that state as long as you created a new issue with all the remaining changes to do on the new SDD design and you decided which one of @Math-R @kmer2016 or you handle it on this PI.

The changes I can see between the two mockups and which should be in the issue AC : 
- no more units on both axis (km/h and km)
- axis x now overflows on the right of the chart until the board right border (To check with @thibautsailly if it's the same behavior when slopes are not displayed)
- SDD bottom (axis and graduations) now overflows on the board footer
- SDD top now touches the board header
- SDD height is still a bit wrong. That should be fixed (To check with @thibautsailly if the mockup height corresponds to the default height or the minimal one)

Check with @thibautsailly other changes that I might not have spot.

thibautsailly commented:
I'm going to prepare more detailed specs for how the SDD should be displayed in its different states.

maelysLeratRosso commented:
> I think the PR is good enough for now.
> 
> As you can see when comparing the [old SDD UI](https://www.sketch.com/s/2fb064cd-c483-46ac-bd30-72705fd63e75/p/9559F6FC-69EE-4DB8-BE1A-77F60F10C97F/canvas) and the [new one](https://www.sketch.com/s/008d6980-d719-4bdc-bb2a-b4c2dda0dc1e/p/8017F0A2-FC69-4111-A2C6-A38E6D37AB65/canvas), a lot has changed in the UI. Theses changes should be taken into account to match the new scenario page design.
> 
> I'm ok to merge this PR on that state as long as you created a new issue with all the remaining changes to do on the new SDD design and you decided which one of @Math-R @kmer2016 or you handle it on this PI.
> 
> The changes I can see between the two mockups and which should be in the issue AC :
> 
> * no more units on both axis (km/h and km)
> * axis x now overflows on the right of the chart until the board right border (To check with @thibautsailly if it's the same behavior when slopes are not displayed)
> * SDD bottom (axis and graduations) now overflows on the board footer
> * SDD top now touches the board header
> * SDD height is still a bit wrong. That should be fixed (To check with @thibautsailly if the mockup height corresponds to the default height or the minimal one)
> 
> Check with @thibautsailly other changes that I might not have spot.

Seen with @thibautsailly : we will create the corresponding tickets, so this PR can be merged as it is

Uriel-Sautron reviewed (changes requested):
> [!WARNING]
> The board wrapper is not dynamic, we do not have access to the different layers.
<img width="1478" height="503" alt="Screenshot From 2025-08-07 09-39-04" src="https://github.com/user-attachments/assets/b0f233df-a694-4693-bbc7-e7568a163c83" />
For me, PR cannot be merged under these conditions.

Yohh commented:
I agree with @Uriel-Sautron, these layers are not accessible, a fix for this will force us to generate more snapshots for e2e tests.

the snapshots are in the git history, we must be careful not to generate useless ones, which will make the history heavier

SharglutDev commented:
> The board wrapper is not dynamic, we do not have access to the different layers.

Didn't understand what you meant at first since all layers could be properly activated.
But you're right, didn't try to activate multiple layers together (speed limit tags and power restrictions looks fine when activated alone since there is still this gap under the chart).

@iLTmpas to be more specific, the SDD needs to be able to display multiples layers next to each other under the chart (electrical profiles, power restrictions and speed limit tags for now). 
Expected behavior is : when we activate/deactivate a layer the board needs to grow/shrink based on the SDD total height, layers included.
Using the resize grip should modify the chart height only so we can still see all layers.

https://github.com/user-attachments/assets/9a7d7caa-8d58-4fc2-adbc-1bbc7b76af51

Yohh commented:
> (speed limit tags and power restrictions looks fine when activated alone since there is still this gap under the chart).

~~not for every resolutions~~:

<img width="1268" height="550" alt="image" src="https://github.com/user-attachments/assets/e3e1bf06-925e-40cd-9cf8-1828a4763fb7" />

edit: sorry, it's not about resolution, but it's about the real height added by the linear layer

clarani committed:
ui-charts: use internalHeight instead of height in SpeedSpaceChart

Previously, the chart managed its own height, which could vary either when the user enlarged it or when they added/
removed linear layers (dynamicalHeight was private to the chart).

We now need to control the entire height of the graph from the parent component, hence the introduction of internalHeight,
which corresponds to the height of the graph (without the linear layers and with its top and bottom margins) and is used
in the various layers to replace height and dynamicalHeight.

Co-authored-by: Louis <iLTmpas@users.noreply.github.com>
Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
ui-charts: drop useEffect in SettingsPanel

This useEffect was used to update the height of the setting panel. It's not required, and
it will allow us to drop getAdaptiveHeight in the following commits.

Co-authored-by: Louis <iLTmpas@users.noreply.github.com>
Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
ui-charts: drop getLinearLayerMarginTop

Co-authored-by: Louis <iLTmpas@users.noreply.github.com>
Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
ui-charts: drop getAdaptiveHeight

Co-authored-by: Louis <iLTmpas@users.noreply.github.com>
Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: remove section height of ResizableSection

Section height is not useful since it's exactly the same thing as height.

Signed-off-by: Louis Tran <louistran29112003@gmail.com>
Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: use activeBoards to display the SimulationResults' map

Signed-off-by: Clara Ni <clara.ni@outlook.fr>
Signed-off-by: Louis Tran <louistran29112003@gmail.com>

clarani commented:
This PR is ready again, @kmer2016 would you mind take a look ? :)

Maymanaf commented on front/tests/009-op-simulation-settings-tab.spec.ts-snapshots/SpeedSpaceChart-AllSettingsEnabled-chromium-linux.png:
The snapshots changes should be dropped

clarani replied:
Done :white_check_mark:

Uriel-Sautron reviewed (commented):
<img width="910" height="246" alt="Screenshot From 2025-08-25 11-33-25" src="https://github.com/user-attachments/assets/515c5454-c10e-4a76-83c6-185a02988ec0" />

The bottom margin is different from that of the mockup.

Uriel-Sautron commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -39,9 +39,9 @@ import { useScenarioContext } from '../hooks/useScenarioContext';
-const HANDLE_TAB_RESIZE_HEIGHT = 20;
 const HIDDEN_CHART_TOP_HEIGHT = 35;
+const SPEED_SPACE_CHART_HEIGHT = 460;
+const SPEEDSPACECHART_MIN_HEIGHT = 400;
```
```suggestion
const SPEED_DISTANCE_DIAGRAM_HEIGHT = 460;
const SPEEDDISTANCEDIAGRAM_MIN_HEIGHT = 400;
```

Uriel-Sautron commented on front/src/modules/simulationResult/components/SpeedSpaceChart/SpeedSpaceChartContainer.tsx:
```diff
@@ -96,57 +88,24 @@ const SpeedSpaceChartContainer = ({
+      {containerWidth > 0 && (
+        <SpeedSpaceChart
+          width={containerWidth || SPEEDSPACECHART_MIN_HEIGHT}
+          height={heightOfSpeedSpaceChartContainer}
```
```suggestion
          height={heightOfSpeedDistanceDiagramContainer}
```

kmer2016 replied:
Maybe we can use a shorter variable name like `heightOfSDDContainer`

clarani replied:
I renamed it height, since it's obvious it's the height of the diagram container

Uriel-Sautron commented on front/src/modules/simulationResult/styles/simulationResult.scss:
```diff
@@ -21,6 +21,14 @@
   }
 }
 
+.speedspacechart-section {
```
```suggestion
.speed-distance-diagram-section {
```

Uriel-Sautron commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -281,26 +279,36 @@ const SimulationResults = ({
-                  style={{
-                    height: `${speedSpaceChartContainerHeight + HANDLE_TAB_RESIZE_HEIGHT}px`,
-                  }}
+              <div className="speedspacechart-section">
```
```suggestion
              <div className="speed-distance-diagram-section">
```

clarani committed:
front: drop speedspacechart-container class

Not used anymore.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: rename speedSpaceChart by speedDistanceDiagram

The correct name of the SpeedSpaceChart is SpeedDistanceDiagram.
No change in the e2e tests and in ui (will be done later).

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani commented:
> <img alt="Screenshot From 2025-08-25 11-33-25" width="910" height="246" src="https://private-user-images.githubusercontent.com/71732278/481560124-515c5454-c10e-4a76-83c6-185a02988ec0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxMjc4NTgsIm5iZiI6MTc1NjEyNzU1OCwicGF0aCI6Ii83MTczMjI3OC80ODE1NjAxMjQtNTE1YzU0NTQtYzEwZS00YTc2LTgzYzYtMTg1YTAyOTg4ZWMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI1VDEzMTIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE5OWEyYWQzMDZjODczMWYzOWQ2NWJjNDU4YzE2MmMxNjBlYWZhODYxOTM0OTkxYmYzZTA5MzNiNjc2ZDAyZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2maqS35K0Kjcp37sUaAqYLw2Y-xyGfOb5OTrwNP4IyQ">
> 
> The bottom margin is different from that of the mockup.

Yes, I noticed that too but someone told me that the style inside the board will be fixed later and not in this PR (can you confirm @kmer2016 ?).
Moreover, this fix requires more changes in the ui component, and I think this PR can be merged without it, if you agree.

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
       <CurveLayer width={WIDTH_OFFSET} height={HEIGHT_OFFSET} store={store} />
       {store.layersDisplay.speedLimits && (
-        <SpeedLimitsLayer width={adjustedWidthRightAxis} height={height} store={store} />
+        <SpeedLimitsLayer width={adjustedWidthRightAxis} height={internalHeight} store={store} />
```
Now, we can use pre-calculated values from `chartHeights` in the render function:


```suggestion
        <SpeedLimitsLayer width={adjustedWidthRightAxis} height={chartHeights.mainChartHeight} store={store} />
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
       )}
       {store.layersDisplay.steps && (
-        <StepsLayer width={adjustedWidthRightAxis} height={height} store={store} />
+        <StepsLayer width={adjustedWidthRightAxis} height={internalHeight} store={store} />
```
```suggestion
        <StepsLayer width={adjustedWidthRightAxis} height={chartHeights.mainChartHeight} store={store} />
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
+        <StepsLayer width={adjustedWidthRightAxis} height={internalHeight} store={store} />
       )}
-      <AxisLayerY width={width} height={height} store={store} />
+      <AxisLayerY width={width} height={internalHeight} store={store} />
```
```suggestion
      <AxisLayerY width={width} height={chartHeights.mainChartHeight} store={store} />
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -159,13 +173,13 @@ const SpeedSpaceChart = ({
             setIsMouseHoveringSettingsPanel={setIsMouseHoveringSettingsPanel}
             translations={translations}
             testIdPrefix="settings-panel"
+            setChartHeight={setHeight}
```
What about creating a specific function in `SpeedSpaceChart` to handle height changes? We could pass this function to `SettingsPanel` instead of passing `setChartHeight`.


```typescript
const adjustHeightForLayerChange = useCallback((layerName: string, isCurrentlyActive: boolean) => {
  let adjustment = 0;
  
  switch (layerName) {
    case 'electricalProfiles':
      adjustment = LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT;
      break;
    case 'powerRestrictions':
      adjustment = LINEAR_LAYERS_HEIGHTS.POWER_RESTRICTIONS_HEIGHT;
      break;
    case 'speedLimitTags':
      adjustment = LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT;
      break;
    default:
      adjustment = 0;
  }
  
  setHeight(prevHeight => prevHeight + (isCurrentlyActive ? -adjustment : adjustment));
}, [setHeight]);
```

This would make the code easier to understand and maintain because all height calculations would be in one place.

```suggestion
            adjustHeightForLayerChange={adjustHeightForLayerChange}
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/common/SettingsPanel.tsx:
```diff
@@ -76,6 +67,26 @@ const SettingsPanel = ({
+                      (store.layersDisplay[selection] ? -1 : 1);
+                  }
+                  return newHeight;
+                });
```
Please check the suggestion [below](https://github.com/OpenRailAssociation/osrd/pull/12594/files#r2298406345) about creating an `adjustHeightForLayerChange` function in the `SpeedSpaceChart` component first:
```suggestion
                adjustHeightForLayerChange(selection, store.layersDisplay[selection]);
```

https://github.com/OpenRailAssociation/osrd/pull/12594/files#r2298406345

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
         <ElectricalProfileLayer
           width={adjustedWidthRightAxis}
-          height={height + LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT}
+          height={internalHeight + LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT}
```
```suggestion
          height={chartHeights.electricalProfileLayerHeight}
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
         <PowerRestrictionsLayer
           width={adjustedWidthRightAxis}
-          marginTop={getLinearLayerMarginTop(height, store.layersDisplay)}
+          marginTop={
+            internalHeight -
+            MARGINS.MARGIN_BOTTOM +
+            (store.layersDisplay.electricalProfiles
+              ? LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT
+              : 0)
+          }
```
```suggestion
          marginTop={chartHeights.powerRestrictionsTop}
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
+      <TickLayerX width={adjustedWidthRightAxis} height={height} store={store} />
       {store.layersDisplay.declivities && (
-        <TickLayerYRight width={width} height={height} store={store} />
+        <TickLayerYRight width={width} height={internalHeight} store={store} />
```
```suggestion
        <TickLayerYRight width={width} height={chartHeights.mainChartHeight} store={store} />
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
           width={adjustedWidthRightAxis}
-          height={dynamicHeight}
-          heightOffset={dynamicHeightOffset}
+          internalHeight={internalHeight}
```
```suggestion
          internalHeight={chartHeights.mainChartHeight}
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
       <FrontInteractivityLayer
         width={WIDTH_OFFSET}
-        height={dynamicHeightOffset}
+        height={
+          internalHeight -
+          MARGINS.MARGIN_BOTTOM -
+          MARGINS.MARGIN_TOP +
+          (store.layersDisplay.electricalProfiles
+            ? LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT
+            : 0) +
+          (store.layersDisplay.powerRestrictions
+            ? LINEAR_LAYERS_HEIGHTS.POWER_RESTRICTIONS_HEIGHT
+            : 0) +
+          (store.layersDisplay.speedLimitTags ? LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT : 0)
+        }
```
```suggestion
        height={chartHeights.interactivityLayerHeight}
```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -92,13 +92,31 @@ const SpeedSpaceChart = ({
+    }
+    return newHeight;
+  }, [height, store.layersDisplay]);
+
```
Calculates all height-related values for the chart based on visible layers. This centralizes all height calculations in one place:

```suggestion
  const chartHeights = useMemo(() => {
  // Calculate offsets for each layer
  const electricalProfilesOffset = store.layersDisplay.electricalProfiles 
    ? LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT 
    : 0;
    
  const powerRestrictionsOffset = store.layersDisplay.powerRestrictions
    ? LINEAR_LAYERS_HEIGHTS.POWER_RESTRICTIONS_HEIGHT
    : 0;
  
  const speedLimitTagsOffset = store.layersDisplay.speedLimitTags
    ? LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT
    : 0;
  
  // Calculate main chart height (equivalent to internalHeight)
  const mainChartHeight = height - (
    electricalProfilesOffset + powerRestrictionsOffset + speedLimitTagsOffset
  );
  
  // Calculate positions for linear layer using the offsets
  
  // Base position for electrical profiles (and other layers when stacked)
  const baseLayerPosition = mainChartHeight - MARGINS.MARGIN_BOTTOM;
  
  // Position for power restrictions layer (needs to account for electrical profiles if visible)
  const powerRestrictionsTop = baseLayerPosition + electricalProfilesOffset;
  
  // Position for speed limit tags layer (needs to account for both previous layers if visible)
  const speedLimitTagsTop = powerRestrictionsTop + powerRestrictionsOffset;
  
  // Calculate specialized heights needed by electricalProfileLayer and interactivityLayer components
  const electricalProfileLayerHeight = mainChartHeight + electricalProfilesOffset;
  
  const interactivityLayerHeight = mainChartHeight - MARGINS.MARGIN_BOTTOM - 
  MARGINS.MARGIN_TOP + 
  electricalProfilesOffset + powerRestrictionsOffset + speedLimitTagsOffset;
  
  return {
    mainChartHeight,
    
    // Linear layers positions
    baseLayerPosition,
    powerRestrictionsTop,
    speedLimitTagsTop,
    
    // Other layers
    electricalProfileLayerHeight,
    interactivityLayerHeight
  };
}, [height, store.layersDisplay]);

```

clarani replied:
Done

kmer2016 commented on front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx:
```diff
@@ -174,52 +188,73 @@ const SpeedSpaceChart = ({
+            (store.layersDisplay.powerRestrictions
+              ? LINEAR_LAYERS_HEIGHTS.POWER_RESTRICTIONS_HEIGHT
+              : 0)
+          }
```
```suggestion
          marginTop={chartHeights.speedLimitTagsTop}
```

clarani replied:
Done

kmer2016 commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -281,26 +279,36 @@ const SimulationResults = ({
+                    </div>
+                  </BoardWrapper>
+                </ResizableSection>
               </div>
```
When the SDD is turned off, the reserved space remains visible even though the component itself disappears. The following lines can fix it.


```suggestion
              <div className="speed-distance-diagram-section">
                {activeBoards.has('sdd') && (
                  <ResizableSection
                    height={SDDHeight}
                    setHeight={setSDDHeight}
                    minHeight={SDD_MIN_HEIGHT}
                  >
                    <BoardWrapper name={t('simulationResults.speedDistanceDiagram')}>
                      <div className="osrd-simulation-container">
                        <div className="chart-container">
                          <SpeedDistanceDiagramWrapper
                            timetableItemSimulation={simulationResults.simulation}
                            selectedTimetableItemPowerRestrictions={
                              simulationResults.powerRestrictions
                            }
                            rollingStock={simulationResults.rollingStock}
                            pathProperties={simulationResults.pathProperties}
                            height={SDDHeight}
                            setHeight={setSDDHeight}
                          />
                        </div>
                      </div>
                    </BoardWrapper>
                  </ResizableSection>
                )}
              </div>
```

clarani replied:
Done

kmer2016 reviewed (changes requested):
Great! The wrapper works very well. The component can grow and shrink normally as expected.

However, I think we can improve the code maintainability. 

**Code Improvement Suggestions**

***1. Centralize height calculations***
Currently, height calculations are scattered across different parts of the component.

We can create a single `chartHeights` object that contains all height-related values.

***2. provide a function for height adjustment logic used in `SettingsPanel`***
The height adjustment logic is duplicated between the main component and the SettingsPanel.

What do you think about create a dedicated function that handles all height changes and pass it to the SettingsPanel component.

***Bug Report***
I found a small bug when the user tries to deactivate the SDD via the menu toolbar. When the SDD is turned off, the reserved space remains visible even though the component itself disappears. See the video below

https://github.com/user-attachments/assets/3bc0afc6-d376-4109-85d7-d57fd4adadbb

We can apply the same approach we used for the STD component. Instead of using the hidden prop on the BoardWrapper (which keeps the space allocated), we should move the conditional rendering to the parent level - wrapping the entire ResizableSection with the board visibility condition

clarani committed:
ui-charts: fix SpeedSpaceChart story

The height provided was not dynamic.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

kmer2016 commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -281,26 +279,35 @@ const SimulationResults = ({
-                    heightOfSpeedSpaceChartContainer={speedSpaceChartContainerHeight}
-                    setHeightOfSpeedSpaceChartContainer={setSpeedSpaceChartContainerHeight}
-                  />
+              {!activeBoards.has('sdd') && (
```
```suggestion
              {activeBoards.has('sdd') && (
```

clarani replied:
Done

clarani committed:
front: put speed space chart into board wrapper

Signed-off-by: Louis Tran <louistran29112003@gmail.com>
Signed-off-by: Clara Ni <clara.ni@outlook.fr>

kmer2016 reviewed (approved):
LGTM

Uriel-Sautron reviewed (approved):
Tested

Maymanaf reviewed (approved):
LGTM for e2e tests ✅
